apps/community/backend/cmd/api/main.go

```diff
@@ -31,63 +26,20 @@ type UserCommunities struct {
 	Communities []CommunityConfig
 }
 
-// from https://github.com/Kubuxu/go-ipfs-swarm-key-gen/blob/master/ipfs-swarm-key-gen/main.go
-func KeyGen() string {
-	key := make([]byte, 32)
-	_, err := rand.Read(key)
-	if err != nil {
-		fmt.Println("While trying to read random source:", err)
-	}
-
-	// when writing to swarm.key, add these to the top:
-	// fmt.Println("/key/swarm/psk/1.0.0/")
-	// fmt.Println("/base16/")
-	return hex.EncodeToString(key)
-}
-
 /*
  CreateCommunity: create a node with new peers, serves as default bootstrap for community
  - Create Node
  - Delete all peers
  - Create swarm key and add to swarm.key
  - Run daemon
- - return swarm + address to broadcast
+ - return error, secret key, peerid,swarm address to broadcast
  - this node automatically becomes the bootstrap peer (for now)
  can either use the api returned by createNode or connect to a new client
  TODO: 	create CommunityConfig struct which contains globals for the community like
 		swarm key and name of it and peer ids in it
 */
 func CreateCommunity(name string, path string) (error, string, string, []string) {
-
-	root := os.Getenv("ROOT")
-	// how to get / set env var for root dir?
-	cmdCreate := &exec.Cmd{
-		Path:   root + "/procs/start.sh",
-		Args:   []string{root + "/procs/start.sh", root + "/ipfs/" + path},
-		Stdout: os.Stdout,
-		Stderr: os.Stderr,
-	}
-
-	s := make([]string, 0)
-	if err := cmdCreate.Run(); err != nil {
-		return err, "", "", s
-	}
-
-	bytes, _ := ioutil.ReadFile(root + "/ipfs/" + path + "/config")
-	var data config.Config
-	err := json.Unmarshal(bytes, &data)
-
-	if err != nil {
-		return err, "", "", s
-	}
-
-	// json struct of config : here we can modify it and write back
-
-	key := KeyGen()
-	keyBytes := []byte("/key/swarm/psk/1.0.0/\n/base16/\n" + key + "\n")
-	err = ioutil.WriteFile(root+"/ipfs/"+path+"/swarm.key", keyBytes, 0755)
-
-	return nil, key, data.Identity.PeerID, data.Addresses.Swarm
+	return ipfs.NewCommunityIPFSNode(name, path)
```

Comment:
Same idea applies for connecting to a community. I think we should have a separate command for creating a new IPFS net, and have the regular ipfs command just be for joining. Since we are going to combine this with the `habitatctl community` commands I created, I think we can wait on these comments for now, since we'd eventually just want to call community specific stuff rather than just IPFS specific stuff